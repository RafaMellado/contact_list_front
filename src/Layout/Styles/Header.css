.header {
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: var(--header-height);
  padding: 1rem;
  box-shadow: -1px 3px 5px 3px var(--bs-gray-200);
  background-color: var(--bs-body-bg);
  z-index: 1;
}
