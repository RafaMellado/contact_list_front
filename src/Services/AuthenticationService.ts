import Cookies from "universal-cookie";
import { LoginRequestBody, LoginResponse } from "./Interfaces/Login";
import { SignUpRequestBody, SignUpResponse } from "./Interfaces/SignUp";
import LoginRepository from "./Repository/LoginRepository";
import SignUpRepository from "./Repository/SignUpRepository";

class AuthenticationService {
  async login(data: LoginRequestBody): Promise<void> {
    const response = await LoginRepository.login(data);

    if (response.token) {
      this.setAuthCookie(response);

      window.location.reload();
    }
  }

  async signUp(data: SignUpRequestBody): Promise<SignUpResponse> {
    return SignUpRepository.signUp(data);
  }

  logout(): void {
    const cookies = new Cookies();

    cookies.remove("contact-list-user");

    window.location.href = "/";
  }

  isLogged(): Boolean {
    const cookies = new Cookies();

    return !!cookies.get("contact-list-user");
  }

  setAuthCookie(data: LoginResponse): void {
    const cookies = new Cookies();

    cookies.set(
      "contact-list-user",
      { token: data.token, username: data.username },
      { expires: new Date(data.exp) }
    );
  }
}

export default new AuthenticationService();
