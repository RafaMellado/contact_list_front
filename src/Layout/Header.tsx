import AuthenticationService from "../Services/AuthenticationService";
import { MenuLogged } from "./Components/MenuLogged";
import { MenuUnlogged } from "./Components/MenuUnlogged";

export function Header() {
  const TRANSLATIONS = "layoutHeader";

  return (
    <div className="position-fixed fixed-top p-3">
      <div className="d-flex justify-content-end">
        {AuthenticationService.isLogged() ? (
          <MenuLogged translations={TRANSLATIONS} />
        ) : (
          <MenuUnlogged translations={TRANSLATIONS} />
        )}
      </div>
    </div>
  );
}
