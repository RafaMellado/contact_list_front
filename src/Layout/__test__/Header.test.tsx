import { Header } from "../Header";
import { act, render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import AuthenticationService from "../../Services/AuthenticationService";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

jest.mock("../../Services/AuthenticationService", () => ({
  isLogged: jest.fn(),
}));

describe(`<${Header.name} />`, () => {
  const history = createMemoryHistory();
  const factoryComponent = () =>
    render(
      <Router location="/" navigator={history}>
        <Header />
      </Router>
    );

  test("renders correctly the component", () => {
    const component = factoryComponent();

    act(() => {
      expect(component.asFragment()).toMatchSnapshot();
    });
  });

  test("renders logged menu header when is logged", () => {
    AuthenticationService.isLogged = () => true;

    const component = factoryComponent();

    act(() => {
      expect(component.asFragment()).toMatchSnapshot();
    });
  });
});
