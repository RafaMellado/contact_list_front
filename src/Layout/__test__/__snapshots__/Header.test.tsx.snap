// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly the component 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <div
      class="d-flex justify-content-end"
    >
      <button
        class="me-3 btn btn-primary"
        data-testid="menu-unlogged-btn"
        type="button"
      >
        layoutHeader.signUpBtn
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Header /> renders logged menu header when is logged 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <div
      class="d-flex justify-content-end"
    >
      <button
        class="me-3 btn btn-primary"
        data-testid="menu-logged-logout-btn"
        type="button"
      >
        layoutHeader.logout
      </button>
    </div>
  </div>
</DocumentFragment>
`;
